src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -365,13 +365,23 @@ bool Terminal::EraseInDisplay(const DispatchTypes::EraseType eraseType)
 
 bool Terminal::SetWindowTitle(std::wstring_view title)
 {
-    _title = title;
-
-    if (_pfnTitleChanged)
+    // Set the title on Terminal load
+    if (_title.empty())
     {
+        _title = title;
         _pfnTitleChanged(title);
     }
 

```

Comment:
I still believe you can remove all this, since the _title should never be empty now that UpdateSettings is taking care of it on startup

Reply:
Removing this causes the tab title to be blank until re-focused on when suppressApplicationTitle is set to true.

Reply:
Hah! Just kidding, just had a slow-to-load tab :)

Reply:
Wait, no. It happens if I haven't previously loaded that profile in another tab. If I have two of the same profiles open, it behaves fine.

Reply:
That shouldn't be the case ... o_O the title should never be empty because it was getting set by `updateSettings`

oh crap i just realized i made you make a bad/annoying change

Reply:
okay, if you roll back to commit 5badf2c you'll be fine. If you don't have any local changes you care about:

```
git reset --hard 5badf2c 
git push -f
```